src/snowflake/cli/_app/telemetry.py

```diff
@@ -81,6 +86,37 @@ class TelemetryEvent(Enum):
 TelemetryDict = Dict[Union[CLITelemetryField, TelemetryField], Any]
 
 
+def _is_cli_exception(exception: Exception) -> bool:
+    return isinstance(
+        exception,
+        (click.ClickException, typer.Exit, typer.Abort, BrokenPipeError),
+    )
+
+
+def _get_additional_exception_information(exception: Exception) -> TelemetryDict:
+    """
+    Attach the errno and sqlstate if the exception or the
+    cause of the exception is a ProgrammingError
+    """
+    additional_info = {}
+
+    if isinstance(exception, ProgrammingError):
+        additional_info[CLITelemetryField.ERROR_CODE] = exception.errno
+        additional_info[CLITelemetryField.SQL_STATE] = exception.sqlstate
```

Comment:
What is `sqlstate`? Is it enum? Any chances of passing sensitive data under this key?